app/main.py

```diff
@@ -1,3 +1,15 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    local_cache = {}
+
+    def wrapper(*args) -> int:
+        if args in local_cache:
+            print("Getting from cache")
+            return local_cache[args]
+        else:
+            local_cache[args] = func(*args)
+            print("Calculating new result")
+            return local_cache[args]
```

Comment:
remove duplicate of code(read checklist for example)
line 10 and 14 the same and can be moved from if-else